src/CloneList.sol

```diff
@@ -30,43 +27,47 @@ abstract contract CloneList {
     }
 
     function popListIndex(uint protoId, uint index) internal {
-        unchecked { // if clone deoesn't exist an error will throw above. should not underflow
-            protoIdToDepth[protoId]--; // decrement clone depth counter
-        }
-        if (index == protoIdToIndexHead[protoId]) { // if index == indexHead move head to next index
-            // index -> next
-            // head = next
-            protoIdToIndexHead[protoId] = protoIdToIndexToAfter[protoId][index];
-        }
+        uint prevIndex = protoIdToIndexToPrior[protoId][index];
+        uint nextIndex = protoIdToIndexToAfter[protoId][index];
         // index pointers will change:
         // prev -> index -> next
         // becomes:
         // prev ----------> next
-        protoIdToIndexToAfter[protoId][protoIdToIndexToPrior[protoId][index]] = protoIdToIndexToAfter[protoId][index];
+        protoIdToIndexToAfter[protoId][prevIndex] = nextIndex;
 
         // prev <- index <- next
         // becomes:
         // prev <---------- next
-        protoIdToIndexToPrior[protoId][protoIdToIndexToAfter[protoId][index]] = protoIdToIndexToPrior[protoId][index];
+        protoIdToIndexToPrior[protoId][nextIndex] = prevIndex;
```

Comment:
my comment somehow disappeared:

> commenting a few things which initially made me hesistant to do these changes. We can reason out if that's ever the case:
this will be wrong if protoIdToIndexToAfter[protoId][protoIdToIndexToPrior[protoId][index]] == protoIdToIndexToAfter[protoId][index] when popListIndex is called.

ie `protoIdToIndexToPrior[protoId][index] == index`.

Reply:
as per my reasoning, this case isn't possible.